cnf-tests/testsuites/e2esuite/ptp/ptp.go

```diff
@@ -3,157 +3,55 @@ package ptp
-
-	"k8s.io/utils/pointer"
-)
-
-const (
-	ptpLinuxDaemonNamespace  = "openshift-ptp"
-	ptpSlaveNodeLabel        = "ptp/test-slave"
-	ptpGrandmasterNodeLabel  = "ptp/test-grandmaster"
-	ptpGrandMasterPolicyName = "test-grandmaster"
-	ptpSlavePolicyName       = "test-slave"
-	ptpDaemonsetName         = "linuxptp-daemon"
 )
 
 var isSingleNode bool
 
 var _ = Describe("[ptp]", func() {
-
+	var fullConfig testconfig.TestConfig
 	execute.BeforeAll(func() {
-		var err error
-		isSingleNode, err = nodes.IsSingleNodeCluster()
-		Expect(err).ToNot(HaveOccurred())
-
-		if !discovery.Enabled() {
-			if isSingleNode {
-				Skip("At least two nodes are required to configure ptp tests. To use an external grandmaster please use discovery mode.")
-			}
-			Clean()
-			ptpNodes, err := nodes.PtpEnabled(client.Client)
-
-			Expect(err).ToNot(HaveOccurred())
-			Expect(len(ptpNodes)).To(BeNumerically(">", 1), "need at least two nodes with ptp capable nics")
-
-			By("Labeling the grandmaster node")
-			ptpGrandMasterNode := ptpNodes[0]
-			ptpGrandMasterNode.NodeObject, err = nodes.LabelNode(ptpGrandMasterNode.NodeName, ptpGrandmasterNodeLabel, "")
-			Expect(err).ToNot(HaveOccurred())
-
-			By("Labeling the slave node")
-			ptpSlaveNode := ptpNodes[1]
-			ptpSlaveNode.NodeObject, err = nodes.LabelNode(ptpSlaveNode.NodeName, ptpSlaveNodeLabel, "")
-			Expect(err).ToNot(HaveOccurred())
-
-			By("Creating the policy for the grandmaster node")
-			err = createConfig(ptpGrandMasterPolicyName,
-				ptpGrandMasterNode.InterfaceList[0],
-				"-2",
-				"-a -r -r",
-				ptpGrandmasterNodeLabel,
-				pointer.Int64Ptr(5))
-			Expect(err).ToNot(HaveOccurred())
-
-			By("Creating the policy for the slave node")
-			err = createConfig(ptpSlavePolicyName,
-				ptpSlaveNode.InterfaceList[0],
-				"-s -2",
-				"-a -r",
-				ptpSlaveNodeLabel,
-				pointer.Int64Ptr(5))
-			Expect(err).ToNot(HaveOccurred())
-
-			By("Restart the linuxptp-daemon pods")
-			ptpPods, err := client.Client.Pods(ptpLinuxDaemonNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
-			Expect(err).ToNot(HaveOccurred())
-
-			for _, pod := range ptpPods.Items {
-				err = client.Client.Pods(ptpLinuxDaemonNamespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{GracePeriodSeconds: pointer.Int64Ptr(0)})
-				Expect(err).ToNot(HaveOccurred())
-			}
+		client.Client = client.New("")
+		l2lib.GlobalL2DiscoveryConfig.SetL2Client(k8sclient.Client, client.Client.Config)
+		testconfig.CreatePtpConfigurations()
+		fullConfig = testconfig.GetFullDiscoveredConfig(utils.PtpLinuxDaemonNamespace, false)
+		if fullConfig.Status != testconfig.DiscoverySuccessStatus {
+			fmt.Printf(`ptpconfigs were not properly discovered, Check:
+- the ptpconfig has a %s label only in the recommend section (no node section)
+- the node running the clock under test is label with: %s`, utils.PtpClockUnderTestNodeLabel, utils.PtpClockUnderTestNodeLabel)
+			os.Exit(1)
```

Comment:
The exit was added to have a cleaner exit when the tool is used on cli but here it will interrupt the test. @aneeshkp can we go back and just fail the test here?

Reply:
fixed